@here/harp-mapview/lib/MapViewTaskScheduler.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { Task, TaskQueue } from "@here/harp-utils";
+
+import { MapView, TileTaskGroups } from "./MapView";
+
+export class MapViewTaskScheduler {
+    private m_taskQueue: TaskQueue;
+    private m_throttlingEnabled: boolean = false;
+
+    constructor(private m_mapView: MapView) {
+        this.m_taskQueue = new TaskQueue({
+            groups: [TileTaskGroups.FETCH_AND_DECODE, TileTaskGroups.CREATE],
+            prioSortFn: (a: Task, b: Task) => {
+                return a.getPrio() - b.getPrio();
+            }
+        });
+    }
+
+    get taskQueue() {
+        return this.m_taskQueue;
+    }
+
+    get throttlingEnabled(): boolean {
+        return this.m_throttlingEnabled === true;
+    }
+
+    set throttlingEnabled(enabled: boolean) {
+        this.m_throttlingEnabled = enabled;
+    }
+
+    /**
+     * Processes the pending Tasks of the underlying [[TaskQueue]]
+     * !! This should run at the end of the renderLoop, so the calculations of the available
+     * frame time are better estimated
+     *
+     * @param frameStartTime the start time of the current frame, is used to calculate the
+     * still available time in the frame to process Tasks
+     *
+     */
+    processPending(frameStartTime: number) {
```

Comment:
Please also add some numbers to the performance statistics, e.g.: Number of tasks processed & number of tasks pending

Reply:
done